code/backend/batch/utilities/helpers/AzureBlobStorageClient.py

```diff
@@ -101,16 +103,30 @@ def file_exists(self, file_name):
 
         return blob_client.exists()
 
-    def upload_file(self, bytes_data, file_name, content_type="application/pdf"):
+    def upload_file(self, bytes_data, file_name, content_type: Optional[str] = None):
         # Create a blob client using the local file name as the name for the blob
         blob_client = self.blob_service_client.get_blob_client(
             container=self.container_name, blob=file_name
         )
+
+        content_settings = ContentSettings(content_type=content_type)
+
+        if content_type is None:
+            content_type = mimetypes.MimeTypes().guess_type(file_name)[0]
+            charset = (
+                f"; charset={chardet.detect(bytes_data)['encoding']}"
+                if content_type == "text/plain"
+                else ""
+            )
+            content_type = content_type if content_type is not None else "text/plain"
+            content_settings = ContentSettings(content_type=content_type + charset)
+
         # Upload the created file
         blob_client.upload_blob(
             bytes_data,
             overwrite=True,
-            content_settings=ContentSettings(content_type=content_type),
+            content_settings=content_settings,
+            metadata={"title": file_name},
```

Comment:
I would think it would be better to make `metadata` an optional parameter and pass it in from where this function is being called in the Admin app, because this function is used in multiple places and we might not always want metadata